packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -149,40 +155,62 @@ export class DetailsService {
     return result;
   }
 
-  public async getLeadersByChainId(
-    chainId?: ChainId,
-    take?: number,
+  public async getLeaders(
+    chainId: ChainId,
+    { orderBy, orderDirection, first }: GetLeadersPaginationOptions = {},
   ): Promise<LeaderDto[]> {
-    const chainIds = !chainId
-      ? await this.networksService.getOperatingNetworks()
-      : [chainId];
+    const filter = this.createLeadersFilter(
+      chainId,
+      orderBy,
+      orderDirection,
+      first,
+    );
+
+    const [rawLeaders, reputations] = await Promise.all([
+      OperatorUtils.getLeaders(filter),
+      this.fetchReputations(
+        chainId,
+        orderBy,
+        orderDirection,
+        orderBy !== LeadersOrderBy.REPUTATION && first,
+      ),
+    ]);
 
-    let allLeadersData: ILeader[] = [];
-    for (const id of chainIds) {
-      const leadersData = await OperatorUtils.getLeaders({ chainId: id });
-      allLeadersData = allLeadersData.concat(
-        leadersData.filter((leader) => leader.amountStaked > 0 && leader.role),
+    const leaders = rawLeaders
+      .filter((leader) => leader.role)
+      .map((leader) =>
+        plainToInstance(LeaderDto, leader, { excludeExtraneousValues: true }),
       );
-    }
 
-    allLeadersData.sort((a, b) =>
-      BigInt(a.amountStaked) >= BigInt(b.amountStaked) ? -1 : 1,
-    );
-    if (take && take > 0) {
-      allLeadersData = allLeadersData.slice(0, take);
-    }
-    const leaders = allLeadersData.map((leader) =>
-      plainToInstance(LeaderDto, leader, {
-        excludeExtraneousValues: true,
-      }),
+    return this.assignReputationsToLeaders(
+      leaders,
+      reputations,
+      orderBy,
+      first,
     );
+  }
 
-    for (const id of chainIds) {
-      const reputations = await this.fetchReputations(id);
-      this.assignReputationsToLeaders(leaders, reputations, id);
-    }
+  private createLeadersFilter(
+    chainId: ChainId,
+    orderBy: LeadersOrderBy,
+    orderDirection: OrderDirection,
+    first?: number,
+  ): ILeadersFilter {
+    const isReputationOrder = orderBy === LeadersOrderBy.REPUTATION;
```

Comment:
Style hint - we could refactor it a bit to improve readability and get rid of many ternaries:
```
const leadersFilter: ILeadersFilter = {
  chainId,
  minAmountStaked: MIN_AMOUNT_STAKED,
  roles: [
    SubgraphOracleRole.JOB_LAUNCHER,
    SubgraphOracleRole.EXCHANGE_ORACLE,
    SubgraphOracleRole.RECORDING_ORACLE,
    SubgraphOracleRole.REPUTATION_ORACLE,
  ],
};

if (orderBy === LeadersOrderBy.REPUTATION) {
  leadersFilter.first = MAX_LEADERS_COUNT;
} else {
  Object.assign(leadersFilter, {
    orderBy,
    orderDirection,
    first,
  });
}

return leadersFilter;
```

Also it will fix `first` value for the case when it's `LeadersOrderBy.REPUTATION` because now if we pass `first = 10` - then we can get to issue with missing leaders, we should get `MAX_LEADERS_COUNT` in this case